day_08.md

```diff
@@ -0,0 +1,188 @@
+>
+> Source: [Wikipedia - Functional Programming](https://en.wikipedia.org/wiki/Functional_programming)
+
+#### Object-oriented programming (OOP)
+
+> Object-oriented programming (OOP) is a programming paradigm based on the concept of "objects", which can contain data, in the form of fields (often known as attributes or properties), and code, in the form of procedures (often known as methods). A feature of objects is an object's procedures that can access and often modify the data fields of the object with which they are associated (objects have a notion of "this" or "self"). In OOP, computer programs are designed by making them out of objects that interact with one another. OOP languages are diverse, but the most popular ones are class-based, meaning that objects are instances of classes, which also determine their types.
+>
+> Source: [Wikipedia - Object-oriented Programming](https://en.wikipedia.org/wiki/Object-oriented_programming)
+
+### Some essential differences
+
+#### Functional
+
+- [High Order Functions (HOF)](https://en.wikipedia.org/wiki/Higher-order_function)
+- [Pure functions](https://en.wikipedia.org/wiki/Pure_function)
+- [Strict](https://en.wikipedia.org/wiki/Eager_evaluation) and [Lazy Evaluation](https://en.wikipedia.org/wiki/Lazy_evaluation)
+- [Referential Transparency](https://en.wikipedia.org/wiki/Referential_transparency)
+- [Recursion](https://en.wikipedia.org/wiki/Recursion_(computer_science)) over [iteration](https://en.wikipedia.org/wiki/Iteration#Computing)
+- [Declarative](https://en.wikipedia.org/wiki/Declarative_programming) over [imperative](https://en.wikipedia.org/wiki/Imperative_programming) programming paradigm
+- [Immutable](https://en.wikipedia.org/wiki/Immutable_object) over mutable data structures
+- [Stateless](https://en.wikipedia.org/wiki/State_(computer_science)#Program_state)
+- [Deterministic](https://en.wikipedia.org/wiki/Deterministic_algorithm)
+
+#### OOP
+
+- [Abstraction](https://en.wikipedia.org/wiki/Abstraction_(computer_science)#Abstraction_in_object_oriented_programming)
+- [Inheritance](https://en.wikipedia.org/wiki/Inheritance_(object-oriented_programming))
+- [Polymorphism](https://en.wikipedia.org/wiki/Polymorphism_(computer_science))
+- [Encapsulation](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming))
+- Iteration over recursion
+- Imperative over declarative programming paradigm
+- Mutable over immutable data structures
+- [Statefull](https://en.wikipedia.org/wiki/State_(computer_science)#Program_state)
+- [Nondeterministic](https://en.wikipedia.org/wiki/Nondeterministic_algorithm)
+
+#### Examples
+
+Let's see this oversimplified example and discuss about it.
```

Comment:
No need for the preposition, should read 'discuss it' instead of 'discuss about it'.